src/components/tables/TableConnected.tsx

```diff
@@ -57,14 +59,48 @@ export const getTableCompositeState = (state: IReactVaporState, id: string): ITa
     } as ITableCompositeState;
 };
 
-const mapStateToProps = (state: IReactVaporState, ownProps: ITableOwnProps): ITableCompositeStateProps => ({
-    tableCompositeState: getTableCompositeState(state, ownProps.id),
-});
+const getDataById = (data: ITableData, props: ITableOwnProps): ITableById => data.byId;
+
+const getSelectedIds = (data: ITableData, props: ITableOwnProps): string[] => data.selectedIds || [];
+
+const getActions = (data: ITableData, props: ITableOwnProps): (rowData?: IData) => IActionOptions[] => props.getActions;
 
-const mapDispatchToProps = (
-    dispatch: IDispatch,
-    ownProps: ITableOwnProps,
-): ITableDispatchProps => ({
+const getMultiSelect = (data: ITableData, props: ITableOwnProps): boolean => props.rowsMultiSelect;
+
+const actionsSelector = createSelector(
+    [getDataById, getSelectedIds, getMultiSelect, getActions],
+    (
+        byId: ITableById,
+        selectedIds: string[],
+        isMultiSelect: boolean,
+        getAction: (rowData?: IData) => IActionOptions[],
+    ): IActionOptions[] => {
+        const rowsData: IData[] = [];

```

Comment:
const rowsData: IData[] = selectedIds.map((id: string) => byId[id]).filter(Boolean);  
  
would do the same as lines 79-84

Reply:
I can use the map but I need my check for the getAction before the filter